bugsnag-android-core/src/androidTest/java/com/bugsnag/android/ClientTest.java

```diff
@@ -327,4 +328,12 @@ public void testUserNotCloned() {
         assertNull(user.getName());
         assertNull(user.getEmail());
     }
+
+    @Test
+    public void testDeviceIdNotUserId() {
+        config.setUser("123", "test@example.com", "Tess Derby");
+        client = new Client(context, config);
+        assertEquals("123", client.getUser().getId());
+        assertNotEquals("123", client.getDeviceDataCollector().generateDevice().getId());
+    }
```

Comment:
Do we have any tests that verify that the user ID is populated from the device ID if no user is set in configuration? If not, can you add one so that requirement is fully tested?

Reply:
Added an additional test case